src/track/track.h

```diff
@@ -81,7 +81,7 @@ class Track : public QObject {
     mixxx::FileInfo getFileInfo() const {
         // Copying mixxx::FileInfo based on QFileInfo is thread-safe due to implicit sharing,
         // i.e. no locking needed.
-        static_assert(sizeof(QSharedDataPointer<int>) == sizeof(mixxx::FileInfo));
+        static_assert(mixxx::FileInfo::isQFileInfo());
```

Comment:
```suggestion
        static_assert(mixxx::FileInfo::canBeCopiedThreadSafe);
```

Reply:
As I have described above, I had this original but picked isQFileInfo because than it is obvious that it is not related to the file itself. It is intended to expose the QFileInfo nature here, because than the developer can read through the QT docs.
It is a difference to expose something to the developer by a name, compared to exposing implementation details to the compiler. The later is now fixed and the first desired.